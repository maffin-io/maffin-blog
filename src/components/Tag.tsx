import React from 'react';
import Link from 'next/link';

type TagProps = {
  text: string,
};

export default function Tag({ text }: TagProps): JSX.Element {
  return (
    <Link href={`/tags/${text}`}>
      <span className="mr-3 text-sm font-medium uppercase text-primary-500 hover:text-primary-600 dark:hover:text-primary-400">
        #
        {text}
      </span>
    </Link>
  );
}
