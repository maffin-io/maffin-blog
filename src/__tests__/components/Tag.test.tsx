import { render } from '@testing-library/react';

import Tag from '@/components/Tag';

describe('Tag', () => {
  it('renders as expected', () => {
    const { container } = render(<Tag text='tag' />);
    expect(container).toMatchSnapshot();
  });
});
