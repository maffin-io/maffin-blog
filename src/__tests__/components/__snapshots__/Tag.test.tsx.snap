// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag renders as expected 1`] = `
<div>
  <a
    href="/tags/tag"
  >
    <span
      class="mr-3 text-sm font-medium uppercase text-primary-500 hover:text-primary-600 dark:hover:text-primary-400"
    >
      tag
    </span>
  </a>
</div>
`;
